### ISSUE TITLE
Issue report: spring-boot-starter-data-cassandra-reactive is actally blocking!

### ISSUE THREAD

**Issue by patpatpat123** [spring-projects/spring-data-examples#577]
Hello team,

While testing a full reactive spring 2.3.2 app with Blockhound, I found this blocking exception.
As part of a reactive stack, there shouldn't be blocking calls, hence opening this issue.
Thank you

Caused by: java.lang.IllegalArgumentException: Error instantiating class AtomicTimestampGenerator (specified by advanced.timestamp-generator.class): Blocking call! java.io.FileOutputStream#writeBytes
	at com.datastax.oss.driver.internal.core.util.Reflection.buildFromConfig(Reflection.java:239)
	at com.datastax.oss.driver.internal.core.util.Reflection.buildFromConfig(Reflection.java:94)
	at com.datastax.oss.driver.internal.core.context.DefaultDriverContext.buildTimestampGenerator(DefaultDriverContext.java:368)
	at com.datastax.oss.driver.internal.core.util.concurrent.LazyReference.get(LazyReference.java:55)
	at com.datastax.oss.driver.internal.core.context.DefaultDriverContext.getTimestampGenerator(DefaultDriverContext.java:743)
	at com.datastax.oss.driver.internal.core.session.DefaultSession$SingleThreaded.init(DefaultSession.java:349)
	at com.datastax.oss.driver.internal.core.session.DefaultSession$SingleThreaded.access$1100(DefaultSession.java:300)
	at com.datastax.oss.driver.internal.core.session.DefaultSession.lambda$init$0(DefaultSession.java:146)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.PromiseTask.run(PromiseTask.java:106)
	at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: reactor.blockhound.BlockingOperationError: Blocking call! java.io.FileOutputStream#writeBytes
	at java.base/java.io.FileOutputStream.writeBytes(FileOutputStream.java)
	at java.base/java.io.FileOutputStream.write(FileOutputStream.java:354)
	at org.apache.logging.log4j.core.appender.OutputStreamManager.writeToDestination(OutputStreamManager.java:250)
	at org.apache.logging.log4j.core.appender.FileManager.writeToDestination(FileManager.java:273)
	at org.apache.logging.log4j.core.appender.rolling.RollingFileManager.writeToDestination(RollingFileManager.java:240)
	at org.apache.logging.log4j.core.appender.OutputStreamManager.flushBuffer(OutputStreamManager.java:282)
	at org.apache.logging.log4j.core.appender.OutputStreamManager.flush(OutputStreamManager.java:291)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.directEncodeEvent(AbstractOutputStreamAppender.java:199)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.tryAppend(AbstractOutputStreamAppender.java:190)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:181)
	at org.apache.logging.log4j.core.appender.RollingFileAppender.append(RollingFileAppender.java:312)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:543)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:502)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:485)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:460)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j.core.Logger.log(Logger.java:161)
	at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2198)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2152)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2135)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:2016)
	at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1875)
	at org.apache.logging.slf4j.Log4jLogger.info(Log4jLogger.java:179)
	at com.datastax.oss.driver.internal.core.time.Clock.getInstance(Clock.java:35)
	at com.datastax.oss.driver.internal.core.time.MonotonicTimestampGenerator.buildClock(MonotonicTimestampGenerator.java:109)
	at com.datastax.oss.driver.internal.core.time.MonotonicTimestampGenerator.<init>(MonotonicTimestampGenerator.java:43)
	at com.datastax.oss.driver.internal.core.time.AtomicTimestampGenerator.<init>(AtomicTimestampGenerator.java:52)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.datastax.oss.driver.internal.core.util.Reflection.buildFromConfig(Reflection.java:232)
	... 14 more

**Comment by mp911de** [comment]
Closing as duplicate of #576.

We use GitHub issues to track bug reports and feature requests. Please post your question on StackOverflow.

**Comment by patpatpat123** [comment]
Hello Mark,

Please help me understand which part of this is NOT a bug report?
Thank you for your great screening

**Comment by mp911de** [comment]
Thanks for asking.

This repository hosts Spring Data Examples. It pulls together dependencies from various sources and doesn't really deal with the internals of the Cassandra driver. Looking at the stack trace, it combines BlockHound, the Cassandra driver and Log4j which ultimately lead to a BlockHound warning. 

One could reason that logging needs to be initialized before starting the driver to avoid blocking calls from within the Cassandra driver. However, with a `RollingFileAppender` it's next to impossible to avoid rollover. On the other side, if you want to avoid file rollover, that's a responsibility of your application configuration.

Furthermore, I'm not even sure that the Cassandra driver guarantees non-blocking internal behavior and logging is a delicate topic when it comes to non-blocking implementations.

Our intent is to share this knowledge with the broader community. Since there's no immediate bug in Spring code here, this issue tracker is the wrong place to discuss your matter. Therefore we ask folks to place their request onto the appropriate forum instead of dropping their requests into places that make it difficult for the community to find a resolution to their issue.

Let me know if this makes it clear why we consider the reported stack trace not a bug.
